### PR TITLE
Fix parser when SCIM begins with a NOT

### LINKED ISSUE #55: SCIM Filter containing AND starting with NOT does not parse correctly
`userType ne "Employee" and not (emails co "example.com" or emails co "example.org")` is a test case in parse.test.ts. I would expect that flipping the two sides of the and would result in the same AST. However, when supplying `not (emails co "example.com" or emails co "example.org") and userType ne "Employee"` the AST that is returned is incorrect.

I will start working on a bug fix, but I am leaving Okta at the end of the month and may not be able to drive this to completion before my last day.

### PR DESCRIPTION
**Pull request by patrickmcdougle-okta**
This currently just adds a failing test case. I will be attempting to fix the bug in this PR as well, but I only have 2 working days at Okta left, so this might have to be taken up by someone else.

Fixes #55

### CODE REVIEW

File: test/parse.test.ts [hunk 1]
```diff
@@ -142,6 +142,16 @@ describe('parse', () =>{
...
     );
+    test(
+      `not (emails co "example.com" or emails co "example.org") and userType ne "Employee"`,
+      and(op("ne", "userType", "Employee"), {
+        op: "not",
+        filter: or(
+          op("co", "emails", "example.com"),
+          op("co", "emails", "example.org")
+        )
+      })
```
Comment (patrickmcdougle-okta) [inline test/parse.test.ts:153]: This is the same AST as in the above test case but with the left and right side of the AND flipped in the SCIM string.

**Comment by patrickmcdougle-okta** [comment]
For the record the CI build failing is expected since I have not attempted to fix the failing test case that I introduced.

**Comment by patrickmcdougle-okta** [comment]
I've been trying to compare this algorithm with the one in https://github.com/pingidentity/scim2/blob/master/scim2-sdk-common/src/main/java/com/unboundid/scim2/common/utils/Parser.java, but I think they are very different strategies. The bad news is that I don't think I'm going to be able to run this to completion as today is my last day at Okta. @patrick-mcdougle might be able to answer questions in his free time, but he might not be able to be 100% dedicated to fixing this on his personal time. Someone else at Okta might be able to pick this up @macyabbey-okta

**Comment by thomaspoignant** [comment]
The fix for the issue was addressed in the #89 pull request.

**Pull request closed without merging**